### ISSUE TITLE
All inlined Phase1 scripts are required to be executed successfully if used in spend or reference inputs

### ISSUE THREAD

**Issue by jmhrpr** [IntersectMBO/cardano-ledger#2826]
In the UTXOW state transition we check that all Phase 1 scripts execute successfully:

`∀s ∈ range(txscripts txw utxo ) ∩ Scriptph1, validateScript s tx`

Implementation: https://github.com/input-output-hk/cardano-ledger/blob/94b1ae3d6b66f4232f34060d89c2f12628998ef2/eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxow.hs#L97-L119

But in the Babbage era `txscripts tx utxo` also includes inline scripts:

```
txscripts tx utxo = txwitscripts tx ∪ refScripts tx utxo

refScripts tx utxo = {hash s 7→ s | ( , , ,s) ∈ utxo (spendInputs tx ∪ refInputs tx)}
```

This means that inlined phase1 scripts in the spend input or reference input UTXOs must always execute successfully even if nothing requires the script. This is contrary to inlined phase2 scripts, which are only executed if they are required in the witnesses.

**Comment by WhatisRT** [comment]
#2827 fixes this in the spec, we now need to fix it in the implementation as well.

**Comment by JaredCorduan** [comment]
closed by  #2833
